pageserver/src/page_cache.rs

```diff
@@ -347,6 +378,11 @@ impl PageCache {
         key: &Key,
         lsn: Lsn,
     ) -> Option<(Lsn, PageReadGuard)> {
+        let Ok(permit) = self.try_get_pinned_slot_permit().await else {
+            // TODO metric?
```

Comment:
It would probably be easier to have the counter on `try_get_pinned_slot_permit` on `Err` arm.

```suggestion
```